web/src/main/java/org/springframework/security/web/authentication/www/GenericAuthenticationFilter.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.web.authentication.www;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter;
+import org.springframework.security.web.util.matcher.AnyRequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+
+/**
+ *
+ * @author Sergey Bespalov
+ *
+ */
+public class GenericAuthenticationFilter extends AbstractAuthenticationProcessingFilter {
```

Comment:
I'd prefer to avoid extending `AbstractAuthenticationProcessingFilter` as it doesn't allow using composition for continuing the `FilterChain` on authentication success

Reply:
fixed

`GenericAuthenticationFilter` extends the `OncePerRequestFilter` and behaves same way as `AuthenticationWebFilter`